lib/libvarnish/vre.c

```diff
@@ -222,40 +240,66 @@ VRE_match(const vre_t *code, const char *subject, size_t length,
 	if (length == 0)
 		length = PCRE2_ZERO_TERMINATED;
 	vre_limit(code, lim);
-	return (vre_match(code, subject, length, 0, options, NULL));
+	return (vre_capture(code, subject, length, 0, options,
+	    NULL, NULL, NULL));
+}
+
+int
+VRE_capture(vre_t *code, const char *subject, size_t length, int options,
+    txt *groups, size_t count, const volatile struct vre_limits *lim)
+{
+	int i;
+
+	CHECK_OBJ_NOTNULL(code, VRE_MAGIC);
+	AN(subject);
+	AZ(options & (~VRE_MASK_MATCH));
+	AN(groups);
+	AN(count);
```

Comment:
for ease of use, do we really need to enforce groups and count?
and, um, yes, I really still think we could just have one _cough_ exec _cough_ function for both.

Reply:
I'd rather cross the exec bridge. We don't need the ovector/txt anywhere in the code base, except in one call site for our regsub implementation. Since we end up with a very different signature if we move to txt, we might as well acknowledge that this is a different function altogether. Even though, I agree, it fulfills the same need.